coral-common/src/main/java/com/linkedin/coral/common/CoralJavaTypeFactoryImpl.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.common;
+
+import java.util.List;
+
+import org.apache.calcite.jdbc.JavaTypeFactoryImpl;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeSystem;
+import org.apache.calcite.rel.type.StructKind;
+
+
+public class CoralJavaTypeFactoryImpl extends JavaTypeFactoryImpl {
+  public CoralJavaTypeFactoryImpl(RelDataTypeSystem typeSystem) {
+    super(typeSystem);
+  }
+
+  @Override
+  public RelDataType createStructType(final List<RelDataType> typeList, final List<String> fieldNameList) {
+    return createStructType(StructKind.PEEK_FIELDS_NO_EXPAND, typeList, fieldNameList);
```

Comment:
how did we choose between `PEEK_FIELDS` vs `PEEK_FIELDS_NO_EXPAND`?

Reply:
lets also capture that in the java doc with examples.

Reply:
I tried to use all three options `PEEK_FIELDS`, `PEEK_FIELDS_DEFAULT` and `PEEK_FIELDS_NO_EXPAND`. The use of  `PEEK_FIELDS` and `PEEK_FIELDS_DEFAULT` would result in the failure of translation of `select *`.  It should be still translated as `select *` in the target SQL, however, it is translated as `select tbl.col1 as col1 .... tbl.coln as coln` when those two options are used. So only `PEEK_FIELDS_NO_EXPAND` is applicable